### ISSUE TITLE
issue with gpb compilation with latest rebar3 

### ISSUE THREAD

**Issue by phani35** [tomas-abrahamsson/gpb#209]
Compilation error is seen when gpb dependency is maintained in _checkouts folder.

```
rebar3_gpb_plugin]$ rebar3 --version
rebar 3.16.1 on Erlang/OTP 23 Erts 11.1.8
```

`rebar3_gpb_plugin]$ rebar3 compile
===> Verifying dependencies...
===> App gpb is a checkout dependency and cannot be locked.
===> Analyzing applications...
===> Compiling gpb
Compiling descriptor.proto...
escript: exception error: undefined function gpb_compile:file/2
===> Hook for compile failed!`

**Comment by tomas-abrahamsson** [comment]
I got from lrascao/rebar3_gpb_plugin#143 how to reproduce, and got from off-gh it could be about rebar3 is now compiling beams to a different directory when there is a `_checkouts` dir.

Could you try out the (temporary) `rebar3.14-_checkouts-compat` branch to see if this works in your context?

**Comment by phani35** [comment]
I have tried with `rebar3.14-_checkouts-compat` branch and it is working good for me.

**Comment by tomas-abrahamsson** [comment]
I merged this into 4.17.7 which I just released, so I'll be closing this issue.
